sktime/utils/_testing/scenarios.py

```diff
@@ -0,0 +1,200 @@
+
+        Returns
+        -------
+        results: output of the last method call, if return_all = False
+            list of deepcopies of all outputs, if return_all = True
+        """
+        # if both None, fill with defaults if exist
+        if method_sequence is None and arg_sequence is None:
+            method_sequence = getattr(self, "default_method_sequence", None)
+            arg_sequence = getattr(self, "default_arg_sequence", None)
+
+        # if only one is None, fill one with the other
+        # this will also raise an error if both are still None
+        if method_sequence is None:
+            method_sequence = _check_list_of_str(arg_sequence)
+        else:
+            method_sequence = _check_list_of_str(method_sequence)
+        if arg_sequence is None:
+            arg_sequence = _check_list_of_str(method_sequence)
+        else:
+            arg_sequence = _check_list_of_str(arg_sequence)
+
+        # check that length of sequences is the same
+        num_calls = len(arg_sequence)
+        if not num_calls == len(method_sequence):
+            raise ValueError("arg_sequence and method_sequence must have same length")
+
+        # execute the commands in sequence, report result(s)
+        results = []
+        for i in range(num_calls):
+            methodname = method_sequence[i]
+            args = self.args[arg_sequence[i]]
+            if methodname != "__init__":
+                res = getattr(obj, methodname)(**args)
+            # if constructor is called, run directly and replace obj
+            else:
+                if isclass(obj):
+                    res = obj(**args)
+                else:
+                    res = type(obj)(**args)
+                obj = res
+            if return_all:
+                results += [deepcopy(res)]
+            else:
+                results = res
+
+        return results
+
+    def is_applicable(self, obj):
+        """Check whether scenario is applicable to obj.
+
+        Abstract method, children should implement. This just returns "true".
+
+        Example for child class: scenario is univariate time series forecasting.
+            Then, this returns False on multivariate, True on univariate forecasters.
+
+        Parameters
+        ----------
+        obj : class or object to check against scenario
+
+        Returns
+        -------
+        applicable: bool
+            True if self is applicable to obj, False if not
+            "applicable" is defined as the implementer chooses, as output of this method
+                False is typically used as a "skip" flag in unit or integration testing
+        """
+        return True
+
+
+def _check_list_of_str(obj, name="obj"):
+    """Check whether obj is a list of str.
+
+    Parameters
+    ----------
+    obj : any object, check whether is list of str
+    name : str, default="obj", name of obj to display in error message
+
+    Returns
+    -------
+    obj, unaltered
+
```

Comment:
why not return None?

Reply:
because then I couldn't follow the `object = check_object(object)` pattern that is used throughout sktime

Reply:
The various checking functions in `datatypes` do not have this interface...? Is that an inconsistency or have I missed something?

Reply:
hm, that's true. I was referring to things like `_check_X` or `_check_fh` or `_check_alpha`, which raise errors instead of returning yes/no. That would be consistent with the `check_raise` function.

The functions starting `check_is` have a different signature, returning boolean, message, metadata.